pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -52,6 +52,7 @@
   private boolean _enableDynamicStarTreeCreation;
   private SegmentPartitionConfig _segmentPartitionConfig;
   private boolean _aggregateMetrics;
+  private List<String> _preAggregationConfigs;
```

Comment:
`String` is too limited. Let's add a class for it so that we can add more properties in the future.

Reply:
done